docs/learn/guides/file-integrations/file-integration-using-ftp/README.md

```diff
@@ -0,0 +1,198 @@
+# File Integration using FTP
+
+The WSO2 FTP Connector lets you connect to an FTP server, and perform operations on files and folders stored on the 
+server. These operations include basic file operations such as reading, updating and deleting files, and listening to 
+the server to invoke operations when a file is created or deleted.
+
+> In this guide you will learn about how to use the WSO2 FTP Connector to create an FTP listener service using Ballerina.
+
+The following are the sections available in this guide.
+
+- [What you'll build](#what-youll-build)
+- [Prerequisites](#prerequisites)
+- [Implementation](#implementation)
+- [Testing](#testing)
+- [Deployment](#deployment)
+
+## What you'll Build
+
+To understand how to build a service to listen to an FTP server, let's consider the use case of a data center which uses 
+an FTP server to store data files. When a new file is added to the server, the FTP listener will read the file and add 
+the file name and size to a map, and when the file is deleted from the server, it will remove the entry from the map. 
+
+## Prerequisites
+ 
+- [Ballerina Distribution](https://ballerina.io/learn/getting-started/)
+- A Text Editor or an IDE 
+> **Tip**: For a better development experience, install one of the following Ballerina IDE plugins: 
+>[VSCode](https://marketplace.visualstudio.com/items?itemName=ballerina.ballerina), 
+>[IntelliJ IDEA](https://plugins.jetbrains.com/plugin/9520-ballerina)
+- An FTP Server (See [here](https://www.digitalocean.com/community/tutorials/how-to-set-up-vsftpd-for-a-user-s-directory-on-ubuntu-16-04) on how to setup an FTP server)
+
+## Implementation
+> If you want to skip the basics, you can download the GitHub repo and directly move to the "Testing" section by skipping  "Implementation" section.
+
+### Create the project structure
+
+Ballerina is a complete programming language that supports custom project structures. Use the following package structure for this guide.
+```
+file-integration-using-ftp
+    ├── src
+       └── ftp_listener
+           └── ftp_listener.bal
+```
+
+Create the Ballerina project `file-integration-using-ftp` and add the `ftp_listener` module using the below commands. 
+
+```bash
+    $ ballerina new file-integration-using-ftp
+    $ cd file-integration-using-ftp
+    $ ballerina add ftp_listener
+```
+
+The above package structure will be created for you. Create the `ftp_listener.bal` file in the Ballerina module.
+
+### Developing the FTP listener service
+
+Let's start implementation by importing the WSO2 FTP Connector in the `ftp_listener.bal` file you just created.
+
+```ballerina
+import wso2/ftp;
```

Comment:
Are we using code directly or pulling code from the file for specific line number range?